cargo-shuttle/src/bacon.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{bail, Result};
+use semver::Version;
+use std::path::Path;
+use tokio::process::Command;
+use tracing::debug;
+
+const MIN_BACON_VERSION: &str = "3.8.0";
+
+#[cfg(unix)]
+const BACON_CONFIG: &str = include_str!("bacon.unix.toml");
+
+#[cfg(windows)]
+const BACON_CONFIG: &str = include_str!("bacon.windows.toml");
+
+/// Runs shuttle in watch mode using bacon
+pub async fn run_bacon(working_directory: &Path) -> Result<()> {
+    check_bacon().await?;
+    debug!("Starting shuttle in watch mode using bacon...");
+
+    Command::new("bacon")
+        .current_dir(working_directory)
+        .args(["-j", "shuttle", "--config-toml", BACON_CONFIG])
+        .stdin(std::process::Stdio::inherit())
+        .stdout(std::process::Stdio::inherit())
+        .stderr(std::process::Stdio::inherit())
+        .spawn()?
+        .wait()
+        .await?
+        .success()
+        .then_some(())
+        .ok_or_else(|| anyhow::anyhow!("bacon process failed"))?;
+
+    Ok(())
+}
+
+async fn check_bacon() -> Result<()> {
+    let output = Command::new("bacon")
+        .arg("--version")
+        .output()
+        .await
+        .map_err(|e| anyhow::anyhow!("Failed to execute bacon: {}\nPlease ensure bacon is installed ('cargo install bacon') and you have the necessary permissions", e))?;
```

Comment:
You can also use context here to add context about the error, and the underlying error will be displayed as a cause, example:

```
Error: Please ensure bacon is installed ('cargo install bacon') and you have the necessary permissions

Caused by:
    No such file or directory (os error 2)
```
```suggestion
        .context("Failed to execute bacon\nPlease ensure bacon is installed ('cargo install bacon') and you have the necessary permissions")?;
```